### PR TITLE
add new classes: HerbivoreProperties, PredatorProperties

### PR DESCRIPTION
**Pull request by RomanPanchuk**
change Animal

### CODE REVIEW

File: src/wildIsland/Main.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package wildIsland;
 
-//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
-// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
-public class Main {
+import wildIsland.Run.Run;
 
+public class Main {
+    public static void main(String[] args) {
+        Run.run();
```
Comment (Volodymyr95) [inline src/wildIsland/Main.java]: why static?

File: src/wildIsland/animals/Animal.java [hunk 2]
```diff
@@ -1,4 +1,153 @@
...
+        this.movingSpeed = movingSpeed;
+    }
+
+    public void move(IslandLocation islandLocation, IslandLocation[][] locations) {
+        islandLocation.getLock().lock();
+        try {
+            int[] newCoordinates = this.getNewCoordinates();
+            if (isLocationFree(islandLocation)) {
+                locations[newCoordinates[0]][newCoordinates[1]].addAnimal(this);
+                if (this instanceof PredatorProperties) {
```
Comment (Volodymyr95) [inline src/wildIsland/animals/Animal.java:33]: don't use instance of, use inheritance

File: src/wildIsland/animals/Animal.java [hunk 3]
```diff
@@ -1,4 +1,153 @@
...
+            } else move(islandLocation, locations);
+        } finally {
+            islandLocation.getLock().unlock();
+        }
+    }
+
+    private boolean isLocationFree(IslandLocation islandLocation) {
+        int maxCount = 0;
+        boolean returnValue;
+        if (this instanceof Boar) {
```
Comment (Volodymyr95) [inline src/wildIsland/animals/Animal.java:48]: don't use it, use inheritance

File: src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            }
+        } finally {
+            location.getLock().unlock();
+        }
+    }
+
+    public void eatPlant(List<Plant> plants, IslandLocation location) {
+        location.getLock().lock();
+        try {
+//            if (plants.size() > 200) {
```
Comment (Volodymyr95) [inline src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java]: don't push commented code

File: src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+                        hunger++;
+                        plants.remove(0);
+                    }
+                    ((Boar) this).setHunger(hunger);
+                }
+                if (this instanceof Buffalo) {
+                    double hunger = ((Buffalo) this).getHunger();
+                    while (hunger < Buffalo.canEat) {
+                        hunger++;
+                        plants.remove(0);
```
Comment (Volodymyr95) [inline src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java]: what if no elements in plants

File: src/wildIsland/islandProperties/IslandLocation.java [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+            }
+        } finally {
+            this.getLock().unlock();
+        }
+    }
+
+    public void startReproduct() {
+        this.getLock().lock();
+        try {
+            for (int i = 0; i < randomReproductCount(Settings.boarMaxCount / Settings.boarBirthRate); i++) {
```
Comment (Volodymyr95) [inline src/wildIsland/islandProperties/IslandLocation.java]: re-write using stream api

File: src/wildIsland/islandProperties/IslandLocation.java [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+                if (herbivores.stream().filter(animal -> animal instanceof Boar).toList().size() > 2) {
+                    herbivores.add(new Boar(x, y));
+                }
+            }
+            for (int i = 0; i < randomReproductCount(Settings.buffaloMaxCount / Settings.buffaloBirthRate); i++) {
+                if (herbivores.stream().filter(animal -> animal instanceof Buffalo).toList().size() > 2) {
+                    herbivores.add(new Buffalo(x, y));
+                }
+            }
+            for (int i = 0; i < randomReproductCount(Settings.caterpillarMaxCount / Settings.caterpillarBirthRate); i++) {
```
Comment (Volodymyr95) [inline src/wildIsland/islandProperties/IslandLocation.java]: why here for all animals?

File: src/wildIsland/Main.java (hunk 1 above)
Comment (RomanPanchuk) [reply src/wildIsland/Main.java]: sorry, fixed it

File: src/wildIsland/animals/Animal.java (hunk 2 above)
Comment (RomanPanchuk) [reply src/wildIsland/animals/Animal.java:33]: I didn't understand what to do

File: src/wildIsland/animals/Animal.java (hunk 3 above)
Comment (RomanPanchuk) [reply src/wildIsland/animals/Animal.java:48]: I didn't understand what to do

File: src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java (hunk 4 above)
Comment (RomanPanchuk) [reply src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java]: deleted

File: src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java (hunk 5 above)
Comment (RomanPanchuk) [reply src/wildIsland/animals/herbivore/properties/HerbivoreProperties.java]: added a condition

File: src/wildIsland/islandProperties/IslandLocation.java (hunk 6 above)
Comment (RomanPanchuk) [reply src/wildIsland/islandProperties/IslandLocation.java]: re-wrote

File: src/wildIsland/islandProperties/IslandLocation.java (hunk 7 above)
Comment (RomanPanchuk) [reply src/wildIsland/islandProperties/IslandLocation.java]: all animals reproduce according to the task

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**